### PR TITLE
Rework MCSCF into the Standard Workflow

### PR DESCRIPTION
**Pull request by fevangelista**
## Description
This PR modifies the way Forte handles frozen core in MSSCF computations and enables the Standard Workflow described in the Forte paper.

## User Notes
- [x] Renamed all `CASSCF_` options to `MCSCF_`. This will require tweaking old input files.
- [x] By default, Forte's standard workflow will run an MCSCF computation using the `ACTIVE_SPACE_SOLVER` provided in the input. This can be avoided by setting `MCSCF_REFERENCE` to `False`.
- [x] By default, the MCSCF code will combine the `FROZEN_DOCC` with the `RESTRICTED_DOCC` orbitals and optimize them. The user can control this with the keyword `MCSCF_FREEZE_CORE` (set by default to `False`).
- [x] After an MCSCF computation, `MCSCF_FREEZE_CORE = FALSE` will mix all inactive orbitals (frozen + restricted doubly occupied) in the semicanonicalization, while `MCSCF_FREEZE_CORE = FALSE` will semicanonicalize them separately.
- [x] Fixed printing of NOs which contained a bug (printed wrong orbital index) 

## Checklist
- [x] Added/updated tests of new features and included a reference `output.ref` file
- [x] Removed comments in code and input files
- [x] Documented source code
- [x] Checked for redundant headers
- [x] Checked for consistency in the formatting of the output file
- [x] Documented new features in the manual
- [x] Ready to go!

### CODE REVIEW

File: forte/orbital-helpers/semi_canonicalize.h [hunk 1]
```diff
@@ -55,9 +55,15 @@ class SemiCanonical {
      * @param quiet_banner Method banner is not printed if set to true
```
Comment (lcyyork) [inline forte/orbital-helpers/semi_canonicalize.h:55]: I guess the comment should be added correspondingly.

File: tests/methods/sa-gasaciscf-1/input.dat [hunk 2]
```diff
@@ -34,14 +34,12 @@ set forte {
   gas1min             [0,0]
   gas1max             [2,1]
   avg_state           [[0,1,1],[2,3,1]]  
-  job_type            casscf #mcscf_two_step
```
Comment (lcyyork) [inline tests/methods/sa-gasaciscf-1/input.dat:37]: The `mcscf_two_step` code should reproduce `casscf` code by setting `CASSCF_MICRO_MAXITER 1`.

Comment (lcyyork) [reply tests/methods/sa-gasaciscf-1/input.dat:37]: I wonder if we should rename the keywords `CASSCF_...` to `MCSCF_...`.

Comment (fevangelista) [reply tests/methods/sa-gasaciscf-1/input.dat:37]: I think we should rename them. Thanks for suggesting this. I think this might be the right PR to do so.

Comment (fevangelista) [reply tests/methods/sa-gasaciscf-1/input.dat:37]: I am writing the documentation for this part of the code, so I'll take care of the renaming and making sure it's documented properly.

File: docs/source/methods/mcscf.rst [hunk 3]
```diff
@@ -132,9 +132,9 @@ Specifically, this is a CASSCF(6,6)/cc-pCVDZ calculation with 2 frozen-core orbi
       active                [2,0,2,2]
       e_convergence         8  # energy convergence of the FCI iterations
       r_convergence         8  # residual convergence of the FCI iterations
-      casscf_e_convergence  8  # energy convergence of the MCSCF iterations
-      casscf_g_convergence  6  # gradient convergence of the MCSCF iterations
-      casscf_micro_maxiter  4  # do at least 4 micro iterations per macro iteration
+      mcscf_e_convergence  8  # energy convergence of the MCSCF iterations
+      mcscf_g_convergence  6  # gradient convergence of the MCSCF iterations
+      mcscf_micro_maxiter  4  # do at least 4 micro iterations per macro iteration
```
Comment (lcyyork) [inline docs/source/methods/mcscf.rst:137]: do at *most* 4 micro iterations per macro iteration

**Comment by fevangelista** [comment]
Ready to merge when test pass.

**Pull request merged**